.github/workflows/ci.yml

```diff
@@ -144,46 +144,46 @@ jobs:
     runs-on: ubuntu-latest
     if: ${{ always() && contains(join(needs.*.result, ','), 'success') }}
     needs: [Linting, Typing, Pytests]
-
+    strategy:
+      matrix:
+        # this line is automatically generated by the script in .github/workflows/scripts/update_notebook_matrix.sh
+        notebook: ["bug_free_python_code.ipynb","check_for_pii.ipynb","competitors_check.ipynb","extracting_entities.ipynb","generate_structured_data.ipynb","generate_structured_data_cohere.ipynb","guardrails_with_chat_models.ipynb","input_validation.ipynb","llamaindex-output-parsing.ipynb","no_secrets_in_generated_text.ipynb","provenance.ipynb","recipe_generation.ipynb","regex_validation.ipynb","response_is_on_topic.ipynb","secrets_detection.ipynb","select_choice_based_on_action.ipynb","streaming.ipynb","syntax_error_free_sql.ipynb","text_summarization_quality.ipynb","toxic_language.ipynb","translation_to_specific_language.ipynb","translation_with_quality_check.ipynb","valid_chess_moves.ipynb","value_within_distribution.ipynb"]
```

Comment:
Is there a way to get these from a script instead of writing them out manually?

Reply:
I've tried! The matrix part is pretty static by Github's specs, best I could do was write the script that generates the matrix.

Reply:
Unfortunate, probably a safe guard so users don't accidentally create an infinite matrix if I had to guess